File: PayFabric/Samples/java/Transaction/Create.java
package com.nodus.payfabric.samples.transaction;

import java.io.BufferedReader;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.URL;

import javax.net.ssl.HttpsURLConnection;

import org.codehaus.jackson.annotate.JsonProperty;

import com.nodus.payfabric.samples.misc.JsonHelper;
import com.nodus.payfabric.samples.misc.Token;

/**
 * This sample is to demo how to create a PayFabric transaction without
 * submitting to payment gateways
 * */
public class Create {

	/**
	 * Create transaction
	 * 
	 * @return New Transaction Key
	 * */
	public String createTransaction() {

		// Populate POST String
		StringBuilder datastring = new StringBuilder();
		datastring.append("{");
		datastring.append("\"Customer\":\"AARONFIT0001\",");
		datastring.append("\"Currency\":\"USD\",");
		datastring.append("\"Amount\":\"4.88\",");
		datastring.append("\"Type\":\"Sale\",");
		//datastring.append("\"Type\":\"Book\",");
		datastring.append("\"SetupId\":\"Paypal\""); // Replace with your gateway account profile name
		datastring.append("}");

		// POST
		try {

			byte[] data = datastring.toString().getBytes("UTF-8");
			String url = "https://sandbox.payfabric.com/payment/api/transaction/create";
			URL obj = new URL(url);
			HttpsURLConnection con = (HttpsURLConnection) obj.openConnection();
			con.setRequestMethod("POST");
			con.setRequestProperty("Content-Type",
					"application/json; charset=utf-8");
			con.setRequestProperty("authorization", new Token().Create());
			con.setRequestProperty("Content-Length", data.length + "");
			con.setDoOutput(true);
			DataOutputStream streamWriter = new DataOutputStream(
					con.getOutputStream());
			streamWriter.write(data);
			streamWriter.flush();
			streamWriter.close();
			InputStream stream;
			int responseCode = con.getResponseCode();
			if (responseCode >= 400) {
				stream = con.getErrorStream();
			} else {
				stream = con.getInputStream();
			}
			BufferedReader streamReader = new BufferedReader(
					new InputStreamReader(stream));
			String inputLine;
			StringBuffer response = new StringBuffer();

			while ((inputLine = streamReader.readLine()) != null) {
				response.append(inputLine);
			}
			streamReader.close();
			con.disconnect();
			System.out.println(response.toString());

			if (responseCode >= 400) {

				// Handling exception from PayFabric
				
			} else {
				TrxKeyResponse trx = JsonHelper.JsonDeserialize(response.toString(), TrxKeyResponse.class);
                return trx.getKey();
			}

		} catch (IOException e) {
			System.out.println(e.getMessage());

			// Handling exception
		}

		return "";
	}

	public static class TrxKeyResponse {
		
		private String key;

		@JsonProperty("Key")
		public String getKey() {
			return key;
		}

		@JsonProperty("Key")
		public void setKey(String key) {
			this.key = key;
		}
		
	}

}

File: PayFabric/Samples/net40/Transaction/Create.cs
// =====================================================================
//  THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
//  KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//  IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//  PARTICULAR PURPOSE.
// =====================================================================

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using System.Net;
using System.Runtime.Serialization;

namespace Samples.Net40
{
    /// <summary>
    /// This sample is to demo how to create a PayFabric transaction without submitting to payment gateways
    /// </summary>
    public partial class Transaction
    {
        /// <summary>
        /// Create transaction
        /// </summary>
        /// <returns>New Transaction Key</returns>
        public string Create()
        {

            try
            {
                //  Populate POST String
                StringBuilder datastring = new StringBuilder();
                datastring.Append("{");
                datastring.Append("\"Customer\":\"AARONFIT0001\",");
                datastring.Append("\"Currency\":\"USD\",");
                datastring.Append("\"Amount\":\"4.88\",");
                datastring.Append("\"Type\":\"Sale\",");
                datastring.Append("\"SetupId\":\"Paypal\""); // Replace with your gateway account profile name
                datastring.Append("}");

                // POST
                byte[] data = System.Text.Encoding.UTF8.GetBytes(datastring.ToString());
                var url = "https://sandbox.payfabric.com/payment/api/transaction/create";
                ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
                HttpWebRequest httpWebRequest = WebRequest.Create(url) as HttpWebRequest;
                httpWebRequest.ContentType = "application/json; charset=utf-8";
                httpWebRequest.Headers["authorization"] = new Token().Create();
                httpWebRequest.Method = "POST";
                httpWebRequest.ContentLength = data.Length;
                Stream stream = httpWebRequest.GetRequestStream();
                stream.Write(data, 0, data.Length);
                stream.Close();
                HttpWebResponse httpWebResponse = httpWebRequest.GetResponse() as HttpWebResponse;
                Stream responseStream = httpWebResponse.GetResponseStream();
                StreamReader streamReader = new StreamReader(responseStream);
                string result = streamReader.ReadToEnd();
                streamReader.Close();
                responseStream.Close();
                httpWebRequest.Abort();
                httpWebResponse.Close();

                // Parse response
                TrxKeyResponse obj = JsonHelper.JsonDeserialize<TrxKeyResponse>(result);
                return obj.Key;
            }
            catch (WebException e)
            {
                //  Handling exception from PayFabric
            }
            catch (Exception e)
            {
                //  Handling exception
            }

            return string.Empty;
        }
    }

    [DataContract]
    public class TrxKeyResponse
    {
        [DataMember]
        public string Key { get; set; }
    }
}
